ivy_tests/test_ivy/helpers.py

```diff
@@ -559,11 +557,23 @@ def test_array_function(
     # assuming value test will be handled manually in the test function
     if not test_values:
         return ret, ret_from_np
-
+    # tolerance dict for dtypes
+    tolerance_dict = {
+        "float16": 1e-02,
+        "float32": 1e-05,
+        "float64": 1e-05,
+        "bfloat16": 1e-02,
+        None: 1e-05,
+    }
+    if not rtol:
+        if ret.dtype in tolerance_dict:
+            rtol = tolerance_dict[ret.dtype]
+        else:
+            rtol = 1e-05
     # flatten the return
     if not isinstance(ret, tuple):
         ret = (ret,)
-    if input_dtype == "bfloat16":
+    if "bfloat16" in input_dtypes:
```

Comment:
This should be called before we compute `ret_from_np`, which will currently throw an error for `bfloat16` data types.